src/zenml/models/v2/core/run_template.py

```diff
@@ -390,4 +409,37 @@ def get_custom_filters(
             )
             custom_filters.append(pipeline_filter)
 
+        if self.user is not None:
+            user_filter = and_(
+                RunTemplateSchema.user_id == UserSchema.id,
+                self.generate_name_or_id_query_conditions(
+                    value=self.user, table=UserSchema
+                ),
+            )
+            custom_filters.append(user_filter)
+
+        if self.pipeline is not None:
+            pipeline_filter = and_(
+                RunTemplateSchema.source_deployment_id
+                == PipelineDeploymentSchema.id,
+                PipelineDeploymentSchema.pipeline_id == PipelineSchema.id,
+                self.generate_name_or_id_query_conditions(
+                    value=self.pipeline,
+                    table=PipelineSchema,
+                ),
+            )
+            custom_filters.append(pipeline_filter)
+
+        if self.stack:
```

Comment:
```suggestion
        if self.stack is not None:
```

Shouldn't this be the same as the two above?